### PR TITLE
Add cache_benchmark

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Towards #676

@clalancette Would you be OK reviewing this?

### CODE REVIEW

File: tf2/test/cache_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Copyright 2024, Willow Garage, Inc. All rights reserved.
```
Comment (ahcorde) [inline tf2/test/cache_benchmark.cpp]:
```suggestion
// Copyright 2024, Open Source Robotics Foundation, Inc.. All rights reserved.
```

File: tf2/test/cache_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      while (timestamp < target_timestamp) {
+        for (int tform_index = 0; tform_index < num_tform; ++tform_index) {
+          example_items[tform_index].frame_id_ = step;
+          example_items[tform_index].stamp_ = timestamp;
+          cache.insertData(example_items[tform_index]);
+        }
+        timestamp += dt;
+        ++step;
+      }
+      return std::make_tuple(timestamp, step);
```
Comment (ahcorde) [inline tf2/test/cache_benchmark.cpp:71]: include `<tuple>`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: tf2/test/cache_benchmark.cpp (hunk 1 above)
Comment (EricCousineau-TRI) [reply tf2/test/cache_benchmark.cpp]: Done c2dd515a

File: tf2/test/cache_benchmark.cpp (hunk 2 above)
Comment (EricCousineau-TRI) [reply tf2/test/cache_benchmark.cpp:71]: Done c2dd515a

File: tf2/test/cache_benchmark.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <benchmark/benchmark.h>
+
+#include <tuple>
```
Comment (clalancette) [inline tf2/test/cache_benchmark.cpp:32]:
```suggestion
#include <chrono>
#include <tuple>
```

**Review by clalancette — CHANGES_REQUESTED** [review]
One more minor change, then I'll run CI on it.

**Comment by EricCousineau-TRI** [comment]
done, thanks!

**Comment by EricCousineau-TRI** [comment]
is it possible to run this through full platform testing? (Linux, Mac, Win) for it to then merge?

**Comment by clalancette** [comment]
> is it possible to run this through full platform testing? (Linux, Mac, Win) for it to then merge?

We don't run CI on macOS, but yes, here is the rest of the CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21084/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15387/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21801/)

**Review by ahcorde — APPROVED** [review]

**Review by clalancette — APPROVED** [review]

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport humble iron

**Pull request merged**